### DISCUSSION TITLE
トラックボールが動きません

### DISCUSSION THREAD

**Discussion by pdichlorobnzn** [Yowkees/keyball#51] [Q&A] [answered]
トラックボールが動かないのですが何を確認すれば良いでしょうか。
・Remapを使って全てのキーは正しく動作していることは確認しています。
・何もキーを押していない状態だとカーソルキーが少しずつ左下へ動いていき、Layer3キーを押すとスクロールアップしているようです。
・トラックボール読み取り基板はキーボードをPCに接続しても暗いままです。

**Comment by Yowkees** [comment, marked as answer]
こんにちは。このたびはKeyball46のご購入、誠にありがとうございます。

ビルドガイドにリンクを貼ったファームウェアをそのまま書き込んだ状態であればソフトには問題ありません。

はんだ付けした部分を中心に再確認をお願いします。
うっかりミスとしてありがちなのが、下記ですので順番にご確認ください。
１．光学センサーICの保護シール２枚のはがし忘れ
２．光学センサーICの方向が逆
３．光学センサーICやProMicroの挿入位置がひとつズレている
４．ICやProMicroのはんだ付け不良
５．ジャンパ８か所のはんだ不良（見た目では問題なくてもテンプラ不良している可能性があるので、もう一度、熱したはんだごてを当ててみて確認

いま一度、上記項目をご確認ください。

＞・トラックボール読み取り基板はキーボードをPCに接続しても暗いままです。
正常に動作していてもセンサーICのレンズ部は肉眼では光って見えませんので、問題ありません。

どうしても直らないようでしたら、こちらではんだ付け済み、動作確認済みのボール読み取り基板と交換対応を行いたいと思います。
ご確認よろしくお願いします。

**Reply by pdichlorobnzn** [reply]
原因は5のジャンパのはんだ付け不良でした。ご指摘の通りもう一度はんだ付けをやり直したところ、無事トラックボールが認識されました。
自分だけでは到底辿りつきそうもないところの問題でしたので、ここで質問させていただき、大変助かりました。
おかげさまでそれ以外にはなんら問題なく動作しています。
あとはお気に入りのキー配列を試行錯誤する楽しみをどっぷり味わいたいと思います。
ありがとうございました。

**Reply by Yowkees** [reply]
無事に動作するようになったようで安心しました。

ぜひKeyball46を楽しんでいただければ幸いです。
今後ともよろしくお願いいたします。

**Comment by kndkndknd** [comment]
Yowkeesさん、こんにちは
このディスカッションの内容に近い内容のため、横から失礼します
（ここではよくないようであれば別にディスカッションを建てさせていただきます）

pdichlorobnznさんと少し事象は異なりますがトラックボールが動かない状況です。
カーソルキーはまったく動きません（keyball_rev1_ball_via.hexを使っています）

> ２．光学センサーICの方向が逆

を一度まさにやってしまっており、はんだを吸い取りつけ直したのですが
その後正しくつけなおしたところ動作しない、という状況なので、
高温をずっと当て続けたためICか基板を壊してしまったのでは、と思っています
（1~5もすべて確認したのですが、駄目でした。。。）

こちらのミスなので非常に申し訳ないのですがボール読み取り基板の交換についてお願いすることは可能でしょうか？

**Reply by Yowkees** [reply]
はんだ付けの失敗による部品の破損をどのように扱うか、キット販売するうえで難しい問題なのですが、
BoothというネットショップにてKeyball46の保守用として、センサーICとレンズや、はんだ付け済み動作確認済みのトラックボール読み取り基板を購入できるように準備いたしました。

下記アドレスサイトをご確認ください。
https://yowkees.booth.pm/

申し訳ありませんが、こちらよりご購入を検討していただけませんでしょうか。
お手数ですがよろしくお願いいたします。

**Reply by kndkndknd** [reply]
ありがとうございます…！早速購入いたしました。
おっしゃるとおり、そこまでのサポートをお願いするのは本来ではないのでは、と迷っていたのですが…対応していただき、本当にありがとうございます。
（対価をどのようにお渡しするかも考えられていなかったので、Boothでお支払い方法含めて作っていただき、感謝しかありません）

いただいた機会を無駄にせず、しっかり完成させてKeyball46の魅力を味わいたいと思います。
本当にありがとうございました。
